src/dynimage.rs

```diff
@@ -758,7 +764,20 @@ fn open_impl(path: &Path) -> ImageResult<DynamicImage> {
         }
     };
 
-    load(fin, format)
+    load(BufReader::new(&fin), format)
+}
+
+/// Open the image located at the path specified.
+/// The image's format is determined from the file header magic bytes.
+pub fn open_magic<P>(path: P) -> ImageResult<DynamicImage>
```

Comment:
With magic bytes detection in `open`, do we need this function?

Reply:
Nope, haha!